tests/components/droplet/conftest.py

```diff
@@ -0,0 +1,97 @@
+"""Common fixtures for the Droplet tests."""
+
+from collections.abc import Generator
+from enum import Enum, auto
+import json
+from unittest.mock import MagicMock, patch
+
+import aiohttp
+import pytest
+
+from homeassistant.components.droplet.const import DOMAIN
+from homeassistant.const import CONF_HOST
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_config_entry() -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: "192.168.1.2"},
+        unique_id="Droplet-1234",
+    )
+
+
+class MockClientBehaviors(Enum):
+    """Options for a mocked WS client."""
+
+    GOOD = auto()
+    FAIL_OPEN = auto()
+    NO_DEVICE_ID = auto()
+
+
+class MockWSConnection:
+    """Mock a websocket connection."""
+
+    def __init__(self, behavior) -> None:
+        """Initialize mock websocket connection."""
+        self.action = behavior
+        self.closed = False
+
+    async def receive(self, timeout):
+        """Mock receiving a message."""
+        match self.action:
+            case MockClientBehaviors.GOOD | MockClientBehaviors.NO_DEVICE_ID:
+                return aiohttp.WSMessage(
+                    aiohttp.WSMsgType.TEXT,
+                    data=json.dumps({"flow": "1.01"}),
+                    extra=None,
+                )
+            case MockClientBehaviors.FAIL_OPEN:
+                return aiohttp.WSMessage(aiohttp.WSMsgType.CLOSED, data="", extra=None)
+
+    async def receive_json(self, timeout):
+        """Mock the receive_json websockets function."""
+        match self.action:
+            case MockClientBehaviors.GOOD:
+                return {"ids": "Droplet-1234"}
+            case _:
+                raise json.JSONDecodeError("error", "", 0)
+
+    async def close(self) -> bool:
+        """Mock closing websocket."""
+        self.closed = True
+        return True
```

Comment:
Instead of mocking a connection like this, mock the library. A cool and simple example on how to mock it for a push based integration, have a look at `knocki`.

This way you can make very versatile and easy to understand tests. If you have questions or need better examples for something that I am not seeing, feel free to push back or send me a message on discord